r/inst/build_arrow_static.sh

```diff
@@ -75,6 +79,7 @@ ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
     -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON \
     -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON \
     -DCMAKE_UNITY_BUILD=${CMAKE_UNITY_BUILD:-ON} \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=${ARROW_LTO:-ON} \
```

Comment:
Should this be `${ARROW_LTO:-$ARROW_DEFAULT_PARAM}`? As it currently is, it looks like LTO will always be turned on.

Reply:
I've changed this to `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=${ARROW_LTO:-OFF}` Confusingly, `$ARROW_DEFAULT_PARAM` here is about adding extra features like compression libraries (based on `$LIBARROW_MINIMAL` above).